src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2794,7 +2794,9 @@ private BoundExpression BindIsOperator(BinaryExpressionSyntax node, DiagnosticBa
                     var hasErrors = node.Right.HasErrors || constantValueOpt is null;
                     isTypeDiagnostics.Free();
                     diagnostics.AddRangeAndFree(isPatternDiagnostics);
-                    var boundConstantPattern = new BoundConstantPattern(node.Right, convertedExpression, constantValueOpt ?? ConstantValue.Bad, operand.Type, hasErrors) { WasCompilerGenerated = true };
+                    var boundConstantPattern = new BoundConstantPattern(
+                        node.Right, convertedExpression, constantValueOpt ?? ConstantValue.Bad, operand.Type, convertedExpression.Type ?? operand.Type, hasErrors)
+                    { WasCompilerGenerated = true };
```

Comment:
>{ WasCompilerGenerated = true }; [](start = 20, length = 32)

Consider shifting initializer to the right, at least on one tab.

Reply:
Will do, and will suppress the resulting warning about whitespace.